templates/certified/search-results.html

```diff
@@ -159,6 +201,21 @@ <h3>Why not try widening your search?</h3>
 </form>
 
 <script>
+  //function to ensure only the option which has been changed is appended to the URL
+  function updateResultsPerPage() {
+    let resultsDropdowns = document.querySelectorAll(".p-results-per-page")
+    resultsDropdowns.forEach(resultsDropdown => {
```

Comment:
I don't think you need 2 loops for this. If you use the `event.target` you may be able to grab the specific `p-results-per-page` HTML element and do the second loop only. Pseudo-code:
In the handler:
```
... onchange=updateResultsPerPage(event) ...
```
```
function (event) {
  let resultsDropdown = event.target;
  const options = resultsDropdown.querySelectorAll("option");
  ....
}
```

Reply:
This doesn't work for this case as within that function I need to be able to access the `resultsDropdown` that has been clicked and also the `resultsDropdown` that hasn't been clicked. The one that hasn't been clicked has to be prevented from being appended to the URL (otherwise 2 x `&limit=...` are added), so therefore I set the `name` attr of the `<select>` which isn't changed to `""` to prevent it being appended when the form is submitted. If you can think of another way to avoid both being appended I open to ideas :)

Reply:
Ok, this sounds like you have to add the `event.preventDefault();` after the `let resultsDropdown = event.target;` Since you are using the same function in both places, it should prevent it from triggering by default on both and then it will only submit when you do it explicitely.

Reply:
This also doesn't work - I think it's because when the form is submitted it checks for which option is selected and appends it to the URL, and because both drop downs have an option selected it adds them both. So even if you prevent default, when the other `resultsDropdown ` is submitted it also submits the selected option from the `resultsDropdown` which isn't interacted with